Navigation/NavigationState.js

```diff
@@ -19,6 +20,11 @@ export type NavigationRoute = string | {
   routes: ?Array<NavigationRoute>;
 };
 
+export const NavigationStateShape = React.PropTypes.shape({
+  routes: React.PropTypes.array.isRequired,
```

Comment:
Maybe make it `arrayOf(React.PropTypes.oneOfType([ React.PropTypes.string, React.PropTypes.object ]))`? Coz things like functions are not valid?